spec/index.bs

```diff
@@ -10808,6 +10892,15 @@ It must only be included by interfaces which also include those mixins.
         The number of draw commands recorded in this encoder.
 </dl>
 
+<div algorithm>
+    To <dfn abstract-op lt="enqueuing a render command">Enqueue a render command</dfn> on
+    {{GPURenderCommandsMixin}} |encoder| which issues the steps of a [=GPU Command=] |command|:
+
+        1. Let |passState| be a snapshot of |encoder|s current state.
+        1. [=list/Append=] |command| to |encoder|.{{GPUCommandsMixin/[[commands]]}}.
+        1. Return |passState|.
+</div>
```

Comment:
(We already talked about this but leaving a comment for the record)

I was going to comment that I prefer this snapshot style rather than keeping state in the queue timeline. (In the queue timeline we don't have a good place to _put_ the state - because state is local to passes, not global to the queue.) BUT: it creates problems because the mixins (which may be on a render bundle encoder) don't actually have access to all the state needed to snapshot. Some options:

- Add an indirection so state is captured at the correct time. I think this means the commands all have to branch depending on whether they're on a pass or bundle encoder, which is messy but doable. Maybe that branch can be factored out.
- Move most render state tracking to queue timeline (put it inside of a render pass state object and unset that object at the end of the pass so it's clear where the state scope ends)
  - Some validation in GPURenderCommandsMixin relies on mutable state, but not all of the states. So we might have to duplicate some between the device timeline and queue timeline but probably not too much?

Reply:
Generally went with the second approach. Stressed for a bit about ways to reduce duplication but none of them felt like the additional indirections and associated obfuscation of what state is available felt worth it. Open to suggestions on how to simplify!